### PR TITLE
SONP-1112 adding basepath as a paramter for upload and download

### PR DESCRIPTION
**Pull request by barry-obrien**

### CODE REVIEW

File: src/main/java/io/swagger/swaggerhub/plugin/SwaggerHubUpload.java [hunk 1]
```diff
@@ -66,6 +66,8 @@
     private String definitionFileNameRegex;
     @Parameter(property = "upload.uploadType", required = true)
     private String uploadType;
+    @Parameter(property = "upload.basepath", required = true)
+    private String basepath;
```
Comment (michaelmelodysmartbear) [inline src/main/java/io/swagger/swaggerhub/plugin/SwaggerHubUpload.java]: I don't think this should be required. They should only add this if required. SaaS users would need to add this and presumably leave it empty. I'm surprised this didn't break tests as test resources don't have this property included.

File: src/test/java/io/swagger/swaggerhub/plugin/SwaggerHubClientTest.java [hunk 2]
```diff
@@ -125,4 +132,76 @@ public void verifySaveIntegrationPluginOfType_postsExpectedRequestBody() throws
...
+                .withRequestBody(matchingJsonPath("$.branch", equalTo(branch)))
+                .withRequestBody(matchingJsonPath("$.enabled", equalTo("true")))
+                .withRequestBody(matchingJsonPath("$.outputFile", equalTo(outputFile)))
+                .withRequestBody(matchingJsonPath("$.repository", equalTo(repository)))
+                .withRequestBody(matchingJsonPath("$.owner", equalTo(repositoryOwner)))
+                .withRequestBody(matchingJsonPath("$.syncMethod", equalTo(syncMethod)))
+                .withRequestBody(matchingJsonPath("$.target", equalTo(target)))
+                .withRequestBody(matchingJsonPath("$.outputFolder", equalTo(outputFolder)))
+                .withRequestBody(matchingJsonPath("$.managedPaths", equalToJson("[\""+outputFile+"\"]")))
+                .withRequestBody(matchingJsonPath("$.name", equalTo(name)));
```
Comment (michaelmelodysmartbear) [inline src/test/java/io/swagger/swaggerhub/plugin/SwaggerHubClientTest.java]: Might be no harm to absract this out to a common method that can be used by each test. For the most part the matcher pattern is the same.

File: src/test/java/io/swagger/swaggerhub/plugin/SwaggerHubClientTest.java [hunk 3]
```diff
@@ -125,4 +132,76 @@ public void verifySaveIntegrationPluginOfType_postsExpectedRequestBody() throws
...
+        boolean enabled = true;
+        String outputFile = "outputFile";
+        String repository = "repository";
+        String repositoryOwner = "repositoryOwner";
+        String scmProvider = "GITHUB";
+        String token = "token";
+        String target = "JSON (Unresolved)";
+        String syncMethod = "Advanced Sync";
+        String name = "Integration Name";
+        String outputFolder = "output/folder";
```
Comment (michaelmelodysmartbear) [inline src/test/java/io/swagger/swaggerhub/plugin/SwaggerHubClientTest.java]: There's a test constants class that you could probably use here instead.

File: src/main/java/io/swagger/swaggerhub/plugin/SwaggerHubUpload.java (hunk 1 above)
Comment (barry-obrien) [reply src/main/java/io/swagger/swaggerhub/plugin/SwaggerHubUpload.java]: Good catch. Thats copy and paste for you. Will look into why this did not break tests though, it should of. If not will try to add a test that verify this.

File: src/test/java/io/swagger/swaggerhub/plugin/SwaggerHubClientTest.java (hunk 2 above)
Comment (barry-obrien) [reply src/test/java/io/swagger/swaggerhub/plugin/SwaggerHubClientTest.java]: I have put this into an abstract method now. We can discuss maybe a better way of doing this as the tests now will be very hard to update if we add more tests to cover more functionality

File: src/main/java/io/swagger/swaggerhub/plugin/SwaggerHubUpload.java (hunk 1 above)
Comment (webron) [reply src/main/java/io/swagger/swaggerhub/plugin/SwaggerHubUpload.java]: There's no try. Only do. 😉

File: src/test/java/io/swagger/swaggerhub/plugin/SwaggerHubClientTest.java (hunk 3 above)
Comment (barry-obrien) [reply src/test/java/io/swagger/swaggerhub/plugin/SwaggerHubClientTest.java]: Have added as constants on top of the class. Will look into creating a constant Test class. Will chat Monday about it. There is a few more we can probably add this also.

**Pull request merged**